lib/compiler/src/sys_core_fold.erl

```diff
@@ -1456,19 +1456,22 @@ clauses(E, [C0|Cs], Ctxt, Sub, LitExpr) ->
 	    [C1];				%Skip the rest
 	{_Mat,no} ->				%Guard fails.
 	    add_warning(C1, nomatch_guard),
-	    clauses(E, Cs, Ctxt, Sub, LitExpr);	%Skip this clause
+	    clauses(E, Cs, Ctxt, Sub, LitExpr, Anno);	%Skip this clause
 	{_Mat,_Suc} ->
-	    [C1|clauses(E, Cs, Ctxt, Sub, LitExpr)]
+	    [C1|clauses(E, Cs, Ctxt, Sub, LitExpr, Anno)]
     end;
-clauses(_, [], _, _, _) -> [].
+clauses(_, [], _, _, _, _) -> [].
 
-shadow_warning([C|Cs], none) ->
+shadow_warning([C|Cs], none, Anno) ->
     add_warning(C, nomatch_shadow),
-    shadow_warning(Cs, none);
-shadow_warning([C|Cs], Line) ->
-    add_warning(C, {nomatch_shadow, Line}),
-    shadow_warning(Cs, Line);
-shadow_warning([], _) -> ok.
+    shadow_warning(Cs, none, Anno);
+shadow_warning([C|Cs], Line, Anno) ->
+    case lists:keyfind('fun', 1, Anno) of
+	{'fun', {Name, Arity}} -> add_warning(C, {nomatch_shadow, Line, {Name, Arity}});
```

Comment:
Long line. Please break it after the `->`. I would break the next line too in the same way.